### DISCUSSION TITLE
Deploy external-secrets to manage secrets in kubernetes

### DISCUSSION THREAD

**Discussion by bengtfredh** [Altinn/altinn-platform#1131] [Ideas]
### Description

Today we already store cert in Azure Key Vault. Cert get downloaded by az cli in pipeline, and added to cluster by terraform. The downside with this is cert is downloaded on an agent filesystem. We only use the agent for 1 run before it get redeployed. We need to run pipeline to deploy new certs in cluster

With Azure Key Vault to Kubernetes (https://akv2k8s.io/) it will check for updated cert on a schedule and download cert to cluster when updated.

- [x] Get akv2k8s approved from security @alt-how 
- [ ] Deploy akv2k8s

### Additional Information

Azure Key Vault to Kubernetes (akv2k8s) makes Azure Key Vault secrets, certificates and keys available in Kubernetes and/or your application - in a simple and secure way.

akv2k8s can have more use cases than only download cert.

**Comment by ghost** [comment]
Looks like an improvement. It is also maintained by Sparebanken Vest.

**Comment by bengtfredh** [comment]
https://external-secrets.io/latest/ may be a better option. It has bigger community with company backing.

**Comment by tjololo** [comment]
For future reference I can add my comment here as well:
I think we should test external-secrets as it seems like it has a bigger community.
I also think we should keep an eye on  [Kubernetes Secrets Store CSI Driver](https://github.com/kubernetes-sigs/secrets-store-csi-driver), but until it is more widely adopted and supported we should stick with the more traditional k8s secret I think.
